src/Microsoft.ML.TimeSeries/STL/Loess.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// Loess is short for Robust Locally Weighted Regression and Smoothing Scatterplots.
+    /// </summary>
+    public class Loess
+    {
+
+        private const double NumericalThreshold = 1.0e-10;
+
+        /// <summary>
+        /// the ratio to determine the local region
+        /// </summary>
+        private readonly int _r;
+
+        private readonly bool _isTemporal;
+
+        /// <summary>
+        /// key is the index of the given point, value is the corresponding neighbors of the given point.
+        /// </summary>
+        private Dictionary<int, LocalRegression> _neighbors;
+
+        private IReadOnlyList<double> _x;
+        private IReadOnlyList<double> _y;
+        private int _length;
```

Comment:
nit: make `_neighbors` and `_length` `readonly`

Reply:
updated.

---
In reply to: [440507244](https://github.com/dotnet/machinelearning/pull/5202#discussion_r440507244) [](ancestors = 440507244)